modules/lang-mustache/src/test/java/org/elasticsearch/script/mustache/MultiSearchTemplateResponseTests.java

```diff
@@ -0,0 +1,95 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.script.mustache;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.action.search.ShardSearchFailure;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.search.internal.InternalSearchResponse;
+import org.elasticsearch.test.AbstractXContentTestCase;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.nullValue;
+
+public class MultiSearchTemplateResponseTests extends AbstractXContentTestCase<MultiSearchTemplateResponse> {
+
+    @Override
+    protected MultiSearchTemplateResponse createTestInstance() {
+        int numItems = randomIntBetween(0, 128);
+        long overallTookInMillis = randomNonNegativeLong();
+        MultiSearchTemplateResponse.Item[] items = new MultiSearchTemplateResponse.Item[numItems];
+        for (int i = 0; i < numItems; i++) {
+            if (randomBoolean()) {
+                // Creating a minimal response is OK, because SearchResponse self
+                // is tested elsewhere.
+                long tookInMillis = randomNonNegativeLong();
+                int totalShards = randomIntBetween(1, Integer.MAX_VALUE);
+                int successfulShards = randomIntBetween(0, totalShards);
+                int skippedShards = totalShards - successfulShards;
+                InternalSearchResponse internalSearchResponse = InternalSearchResponse.empty();
+                SearchResponse.Clusters clusters = new SearchResponse.Clusters(totalShards, successfulShards, skippedShards);
+                SearchTemplateResponse searchTemplateResponse = new SearchTemplateResponse();
+                SearchResponse searchResponse = new SearchResponse(internalSearchResponse, null, totalShards,
+                        successfulShards, skippedShards, tookInMillis, ShardSearchFailure.EMPTY_ARRAY, clusters);
+                searchTemplateResponse.setResponse(searchResponse);
+                items[i] = new MultiSearchTemplateResponse.Item(searchTemplateResponse, null);
+            } else {
+                items[i] = new MultiSearchTemplateResponse.Item(null, new ElasticsearchException("an error"));
+            }
+        }
+        return new MultiSearchTemplateResponse(items, overallTookInMillis);
+    }
+
+    @Override
+    protected MultiSearchTemplateResponse doParseInstance(XContentParser parser) throws IOException {
+        return MultiSearchTemplateResponse.fromXContext(parser);
+    }
+
+    @Override
+    protected boolean supportsUnknownFields() {
+        return false;
```

Comment:
ideally, you would leave this to true and exclude some of the paths where injecting random fields is not supported, I can imagine that the top-level is problematic here and the _source part as well. You can override `getRandomFieldsExcludeFilter`.

Reply:
I tried it and the debugger showed me it failing on an unknown random field name (something like "XCZXFDVC"). Is the end goal here to ensure strictness is being enforced (the test correctly rejects and reports any random noise in the xcontent fields) or that tolerance is applied (the rest client will silently tolerate unknown additions)?

Reply:
that tolerance is applied, it's a way to test forward compatibility of the client.

Reply:
obviously there are places where we injecting random fields makes no sense, like _source etc

Reply:
Makes sense. I don't think `getRandomFieldsExcludeFilter ` will be enough and the parsing code in  `MultiSearchResponse.Item#itemFromXContent` will need changing. Currently it has branching logic like this: 

	 if ("error".equals(fieldName)) {
		item = new Item(null, ElasticsearchException.failureFromXContent(parser));
	} else  ...{
		// assume non-error and parse the hits etc
		item = new Item(SearchResponse.innerFromXContent(parser), null);

The `getRandomFieldsExcludeFilter` only lets me filter on the name but here I would need to filter if the object context being injected into already contains an `error` field as the code assumes that errors are the only field in an error response.